From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Togira <70365614+Togira123@users.noreply.github.com>
Date: Sat, 4 Jun 2022 17:28:48 +0200
Subject: [PATCH] Add VillagerHomeChangeEvent


diff --git a/src/main/java/io/papermc/paper/event/entity/VillagerHomeChangeEvent.java b/src/main/java/io/papermc/paper/event/entity/VillagerHomeChangeEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..a88581248c1611f1a0f4cde4030e2a25fd952a4c
--- /dev/null
+++ b/src/main/java/io/papermc/paper/event/entity/VillagerHomeChangeEvent.java
@@ -0,0 +1,45 @@
+package io.papermc.paper.event.entity;
+
+import org.bukkit.entity.Villager;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.entity.EntityEvent;
+import org.jetbrains.annotations.NotNull;
+
+
+public class VillagerHomeChangeEvent extends EntityEvent implements Cancellable {
+
+    private static final HandlerList handlers = new HandlerList();
+
+    private boolean cancelled;
+
+    public VillagerHomeChangeEvent(@NotNull Villager what) {
+        super(what);
+    }
+
+    @NotNull
+    public Villager getEntity() {
+        return (Villager) entity;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return cancelled;
+    }
+
+    @Override
+    public void setCancelled(boolean cancel) {
+        this.cancelled = cancel;
+    }
+
+    @NotNull
+    @Override
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    @NotNull
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+}
