From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Togira <70365614+Togira123@users.noreply.github.com>
Date: Sat, 4 Jun 2022 17:29:26 +0200
Subject: [PATCH] Add VillagerHomeChangeEvent


diff --git a/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java b/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
index 43243537b765a2d270be6de3f053fea77ff67d18..498018d328e8e5783f0d58e6a388292301ddfc52 100644
--- a/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
+++ b/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
@@ -76,6 +76,17 @@ public class AcquirePoi extends Behavior<PathfinderMob> {
         this.batchCache.long2ObjectEntrySet().removeIf((entry) -> {
             return !entry.getValue().isStillValid(time);
         });
+        // Paper start - VillagerHomeChangeEvent
+        if (this.memoryToAcquire.equals(MemoryModuleType.HOME)) {
+            org.bukkit.entity.Entity bukkitEntity = entity.getBukkitEntity();
+            if (bukkitEntity instanceof org.bukkit.entity.Villager) {
+                boolean cancelled = !new io.papermc.paper.event.entity.VillagerHomeChangeEvent((org.bukkit.entity.Villager) bukkitEntity).callEvent();
+                if (cancelled) {
+                    return;
+                }
+            }
+        }
+        // Paper end - VillagerHomeChangeEvent
         Predicate<BlockPos> predicate = (pos) -> {
             AcquirePoi.JitteredLinearRetry jitteredLinearRetry = this.batchCache.get(pos.asLong());
             if (jitteredLinearRetry == null) {
