From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Togira <70365614+Togira123@users.noreply.github.com>
Date: Sat, 4 Jun 2022 17:34:35 +0200
Subject: [PATCH] Add VillagerMeetingPointChangeEvent


diff --git a/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java b/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
index 498018d328e8e5783f0d58e6a388292301ddfc52..27d5bd9d88df626e0f9945091bd156bc4874e6ad 100644
--- a/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
+++ b/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
@@ -85,6 +85,14 @@ public class AcquirePoi extends Behavior<PathfinderMob> {
                     return;
                 }
             }
+        } else if (this.memoryToAcquire.equals(MemoryModuleType.MEETING_POINT)) {
+            org.bukkit.entity.Entity bukkitEntity = entity.getBukkitEntity();
+            if (bukkitEntity instanceof org.bukkit.entity.Villager) {
+                boolean cancelled = !new io.papermc.paper.event.entity.VillagerMeetingPointChangeEvent((org.bukkit.entity.Villager) bukkitEntity).callEvent();
+                if (cancelled) {
+                    return;
+                }
+            }
         }
         // Paper end - VillagerHomeChangeEvent
         Predicate<BlockPos> predicate = (pos) -> {
