From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Togira <70365614+Togira123@users.noreply.github.com>
Date: Sat, 4 Jun 2022 17:43:54 +0200
Subject: [PATCH] Add VillagerJobSiteChangeEvent


diff --git a/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java b/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
index 27d5bd9d88df626e0f9945091bd156bc4874e6ad..f766f4d0072ae496f0d541dd165cc5350cbe5c87 100644
--- a/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
+++ b/src/main/java/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
@@ -76,7 +76,7 @@ public class AcquirePoi extends Behavior<PathfinderMob> {
         this.batchCache.long2ObjectEntrySet().removeIf((entry) -> {
             return !entry.getValue().isStillValid(time);
         });
-        // Paper start - VillagerHomeChangeEvent
+        // Paper start - VillagerHomeChangeEvent, VillagerMeetingPointChangeEvent & VillagerJobSiteChangeEvent
         if (this.memoryToAcquire.equals(MemoryModuleType.HOME)) {
             org.bukkit.entity.Entity bukkitEntity = entity.getBukkitEntity();
             if (bukkitEntity instanceof org.bukkit.entity.Villager) {
@@ -93,8 +93,16 @@ public class AcquirePoi extends Behavior<PathfinderMob> {
                     return;
                 }
             }
+        } else if (this.memoryToAcquire.equals(MemoryModuleType.JOB_SITE)) {
+            org.bukkit.entity.Entity bukkitEntity = entity.getBukkitEntity();
+            if (bukkitEntity instanceof org.bukkit.entity.Villager) {
+                boolean cancelled = !new io.papermc.paper.event.entity.VillagerJobSiteChangeEvent((org.bukkit.entity.Villager) bukkitEntity).callEvent();
+                if (cancelled) {
+                    return;
+                }
+            }
         }
-        // Paper end - VillagerHomeChangeEvent
+        // Paper end - VillagerHomeChangeEvent, VillagerMeetingPointChangeEvent & VillagerJobSiteChangeEvent
         Predicate<BlockPos> predicate = (pos) -> {
             AcquirePoi.JitteredLinearRetry jitteredLinearRetry = this.batchCache.get(pos.asLong());
             if (jitteredLinearRetry == null) {
